src/ImageSharp/ImageFrame{TPixel}.cs

```diff
@@ -246,27 +246,23 @@ internal ImageFrame<TPixel2> CloneAs<TPixel2>()
                 return this.Clone() as ImageFrame<TPixel2>;
             }
 
-            Func<Vector4, Vector4> scaleFunc = PackedPixelConverterHelper.ComputeScaleFunction<TPixel, TPixel2>();
-
             var target = new ImageFrame<TPixel2>(this.MemoryManager, this.Width, this.Height, this.MetaData.Clone());
 
-            using (PixelAccessor<TPixel> pixels = this.Lock())
-            using (PixelAccessor<TPixel2> targetPixels = target.Lock())
-            {
-                Parallel.For(
-                    0,
-                    target.Height,
-                    Configuration.Default.ParallelOptions,
-                    y =>
+            Parallel.For(
+                0,
+                target.Height,
+                Configuration.Default.ParallelOptions,
+                y =>
+                {
+                    Span<TPixel> sourceRow = this.GetPixelRowSpan(y);
+                    Span<TPixel2> targetRow = target.GetPixelRowSpan(y);
+
+                    for (int x = 0; x < target.Width; x++)
                     {
-                        for (int x = 0; x < target.Width; x++)
-                        {
-                            var color = default(TPixel2);
-                            color.PackFromVector4(scaleFunc(pixels[x, y].ToVector4()));
-                            targetPixels[x, y] = color;
-                        }
-                    });
-            }
+                        ref var pixel = ref targetRow[x];
+                        pixel.PackFromScaledVector4(sourceRow[x].ToScaledVector4());
```

Comment:
The only change I strongly suggest is to introduce new batch primitives for `PackFromScaledVector4()` and `ToScaledVector4()` in `PixelOperations<T>` .
We could drop the for loop then, and push the data through a temporary `Vector4` buffer.

Reply:
I’m not convinced by the need for this. We don’t have it for the existing vector methods and we don’t need any temporary allocations just now.

Reply:
My bad, they do exist. Was looking in the wrong place! Will update now

Reply:
`ParallelFor.WithTemporaryBuffer` could help providing the buffer efficiently.

Reply:
I've pushed the PixelOperations plus tests but in this instance I do not feel that we warrant that allocation at all. We're over-complicating a very simple operation. We're looping in PixelOperations! 

What I have noticed though is that we cannot run a parallel operation within `ImageFrame<T>` using anything other the default configuration `ParallelOptions` values for the action. 

We are passing `MemoryManager` instance to the frame constructor when we should be passing the `Configuration` instance.

Reply:
We need to solve this then, but it could be done in a  separate PR then to minimalize the current changeset.

Reply:
I'll leave a few comments to clarify my point though:

> We're over-complicating a very simple operation. We're looping in PixelOperations!

The point of `PixelOperations<T>` is to allow SIMD optimized path for specific operations. The [optimized path for unscaled Rgba <-> Vector4 packing](https://github.com/SixLabors/ImageSharp/blob/master/src/ImageSharp/PixelFormats/Rgba32.PixelOperations.cs#L39) is about 2x faster than the standard.

> in this instance I do not feel that we warrant that allocation at all

[`ParallelFor.WithTemporaryBuffer`](https://github.com/SixLabors/ImageSharp/blob/master/src/ImageSharp/Common/Helpers/ParallelFor.cs#L30) allows us to solve the above point from only `ParallelOptions.MaxDegreeOfParallelism` number of buffer "allocations" (which are actually pooled array usages!).

Reply:
Ok, forgot about the `Rgba32` optimizations you did. I've made the change and am using the default configuration for now. I'll open another PR for passing the `Configuration` instance